clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -63,69 +64,74 @@ public void configure(Map<String, ?> configs) {}
     @Override
     public void init(List<KafkaMetric> metrics) {
         synchronized (LOCK) {
-            for (KafkaMetric metric : metrics)
-                addAttribute(metric);
-            for (KafkaMbean mbean : mbeans.values())
+            for (KafkaMetric metric : metrics) {
+                final KafkaMbean mbean = addAttribute(metric);
+                if (!this.mbeans.containsKey(mbean.metricName)) {
```

Comment:
I've move all updates for a non-thread-safe `mbeans` map to `synchronized` blocks. So now neither `addAttribute`, nor `getMBean` updates it. In my opinion this improves readability.